### ISSUE TITLE
Qt/5.15.2 Qt/5.12.10: Fails to cross compile using armv7hf

### ISSUE THREAD

**Issue by pbeTrifork** [bincrafters/community#1346]
When trying to cross compile Qt for armv7 the build fails with:
```
Project ERROR: Cannot run target compiler 'arm-linux-gnueabi-g++'. Output:
===================
===================
Maybe you forgot to setup the environment?
```
This problem is consistent for both 5.12 and 5.15

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **Qt/5.12.10** **Qt/5.15.2**
  * Operating System+version: **Linux Ubuntu**
  * Compiler+version: **GCC 7**
  * Docker image: **conanio/gcc7-armv7hf**
  * Conan version: **conan 1.32.1**
  * Python version: **Python 3.7.5**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
Configuration for profile default:

[settings]
os=Linux
os_build=Linux
arch=armv7hf
arch_build=x86_64
compiler=gcc
compiler.version=7
compiler.libcxx=libstdc++
build_type=Release
[options]
[build_requires]
[env]

```

### Steps to reproduce (Include if Applicable)
Options turned off to get Qt working when cross compiling, otherwise fails on flex.

 * `conan remote add bincrafters https://api.bintray.com/conan/bincrafters/public-conan`
 * `conan install qt/5.15.2@bincrafters/stable -o qt:with_icu=False -o qt:GUI=False -o qt:widgets=False -o qt:with_glib=False -o qt:with_mysql=False --build=missing`
 * `conan install qt/5.12.10@bincrafters/stable -o qt:with_icu=False -o qt:GUI=False -o qt:widgets=False -o qt:with_glib=False -o qt:with_mysql=False --build=missing`

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
+ /home/conan/.conan/data/qt/5.15.2/bincrafters/stable/source/qt5/qtbase/configure -top-level -shared -confirm-license -silent -nomake examples -nomake tests -prefix /home/conan/.conan/data/qt/5.15.2/bincrafters/stable/package/9782a352f2b22896a551d487d08d69fcf0feb9c7 -v -opensource -no-gui -no-widgets -release -skip qtsvg -skip qtdeclarative -skip qtactiveqt -skip qtscript -skip qtmultimedia -skip qttools -skip qtxmlpatterns -skip qttranslations -skip qtdoc -skip qtlocation -skip qtsensors -skip qtconnectivity -skip qtwayland -skip qt3d -skip qtimageformats -skip qtgraphicaleffects -skip qtquickcontrols -skip qtserialbus -skip qtserialport -skip qtx11extras -skip qtmacextras -skip qtwinextras -skip qtandroidextras -skip qtwebsockets -skip qtwebchannel -skip qtwebengine -skip qtwebview -skip qtquickcontrols2 -skip qtpurchasing -skip qtcharts -skip qtdatavis3d -skip qtvirtualkeyboard -skip qtgamepad -skip qtscxml -skip qtspeech -skip qtnetworkauth -skip qtremoteobjects -skip qtwebglplugin -skip qtlottie -skip qtquicktimeline -skip qtquick3d --zlib=system -no-opengl -no-vulkan -openssl-linked --glib=no --pcre=system --fontconfig=no --icu=no --sql-mysql=no --sql-psql=yes --sql-odbc=yes --zstd=yes -system-doubleconversion -no-freetype -no-harfbuzz -no-libjpeg -no-libpng -system-sqlite ZLIB_LIBS=-lz OPENSSL_LIBS=-lssl -lcrypto -ldl -lpthread PCRE2_LIBS=-lpcre2-posix -lpcre2-8 -lpcre2-16 -lpcre2-32 -lz -lbz2 DOUBLECONVERSION_LIBS=-ldouble-conversion SQLITE_LIBS=-lsqlite3 -lpthread -ldl PSQL_LIBS=-lpq -lpgcommon -lpthread -lz ODBC_LIBS=-lodbc -lodbcinst -lodbccr -lltdl -lpthread -ldl -liconv -lcharset ZSTD_LIBS=-lzstd -lpthread -I /home/conan/.conan/data/openssl/1.1.1h/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/include -L /home/conan/.conan/data/openssl/1.1.1h/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/lib -I /home/conan/.conan/data/pcre2/10.33/_/_/package/0c7f50fd380a8c3393ca389ff119ec7e28974d9d/include -D PCRE2_STATIC -L /home/conan/.conan/data/pcre2/10.33/_/_/package/0c7f50fd380a8c3393ca389ff119ec7e28974d9d/lib -I /home/conan/.conan/data/double-conversion/3.1.5/_/_/package/da961450d48f172a353d3173ff7efe54d12b811e/include -L /home/conan/.conan/data/double-conversion/3.1.5/_/_/package/da961450d48f172a353d3173ff7efe54d12b811e/lib -I /home/conan/.conan/data/sqlite3/3.31.0/_/_/package/72fab956424a0a2d94c722cc1bfd294beb5cba7c/include -L /home/conan/.conan/data/sqlite3/3.31.0/_/_/package/72fab956424a0a2d94c722cc1bfd294beb5cba7c/lib -I /home/conan/.conan/data/libpq/11.5/_/_/package/f3748ef0523f8bdf5382a8cc0624c2586cc3008e/include -L /home/conan/.conan/data/libpq/11.5/_/_/package/f3748ef0523f8bdf5382a8cc0624c2586cc3008e/lib -I /home/conan/.conan/data/odbc/2.3.7/_/_/package/93d7bafd168ad0429ef9bbb03015255a4f2068f5/include -L /home/conan/.conan/data/odbc/2.3.7/_/_/package/93d7bafd168ad0429ef9bbb03015255a4f2068f5/lib -I /home/conan/.conan/data/zstd/1.4.4/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/include -L /home/conan/.conan/data/zstd/1.4.4/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/lib -I /home/conan/.conan/data/zlib/1.2.11/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/include -L /home/conan/.conan/data/zlib/1.2.11/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/lib -I /home/conan/.conan/data/bzip2/1.0.8/_/_/package/0f605cc66ff048559e2527dae1e478519b0adb51/include -L /home/conan/.conan/data/bzip2/1.0.8/_/_/package/0f605cc66ff048559e2527dae1e478519b0adb51/lib -I /home/conan/.conan/data/libiconv/1.16/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/include -L /home/conan/.conan/data/libiconv/1.16/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/lib -psql_config /home/conan/.conan/data/libpq/11.5/_/_/package/f3748ef0523f8bdf5382a8cc0624c2586cc3008e/bin/pg_config -D_GLIBCXX_USE_CXX11_ABI=0 -xplatform linux-arm-gnueabi-g++ QMAKE_CC=arm-linux-gnueabihf-gcc-7 QMAKE_LINK_C=arm-linux-gnueabihf-gcc-7 QMAKE_LINK_C_SHLIB=arm-linux-gnueabihf-gcc-7 QMAKE_CXX=arm-linux-gnueabihf-g++-7 QMAKE_LINK=arm-linux-gnueabihf-g++-7 QMAKE_LINK_SHLIB=arm-linux-gnueabihf-g++-7
Performing shadow build...
Preparing build tree...
Creating qmake...
Info: creating super cache file /home/conan/.conan/data/qt/5.15.2/bincrafters/stable/build/9782a352f2b22896a551d487d08d69fcf0feb9c7/.qmake.super
Info: creating cache file /home/conan/.conan/data/qt/5.15.2/bincrafters/stable/build/9782a352f2b22896a551d487d08d69fcf0feb9c7/.qmake.cache
Command line: -shared -confirm-license -silent -nomake examples -nomake tests -prefix /home/conan/.conan/data/qt/5.15.2/bincrafters/stable/package/9782a352f2b22896a551d487d08d69fcf0feb9c7 -v -opensource -no-gui -no-widgets -release -skip qtsvg -skip qtdeclarative -skip qtactiveqt -skip qtscript -skip qtmultimedia -skip qttools -skip qtxmlpatterns -skip qttranslations -skip qtdoc -skip qtlocation -skip qtsensors -skip qtconnectivity -skip qtwayland -skip qt3d -skip qtimageformats -skip qtgraphicaleffects -skip qtquickcontrols -skip qtserialbus -skip qtserialport -skip qtx11extras -skip qtmacextras -skip qtwinextras -skip qtandroidextras -skip qtwebsockets -skip qtwebchannel -skip qtwebengine -skip qtwebview -skip qtquickcontrols2 -skip qtpurchasing -skip qtcharts -skip qtdatavis3d -skip qtvirtualkeyboard -skip qtgamepad -skip qtscxml -skip qtspeech -skip qtnetworkauth -skip qtremoteobjects -skip qtwebglplugin -skip qtlottie -skip qtquicktimeline -skip qtquick3d --zlib=system -no-opengl -no-vulkan -openssl-linked --glib=no --pcre=system --fontconfig=no --icu=no --sql-mysql=no --sql-psql=yes --sql-odbc=yes --zstd=yes -system-doubleconversion -no-freetype -no-harfbuzz -no-libjpeg -no-libpng -system-sqlite ZLIB_LIBS=-lz 'OPENSSL_LIBS=-lssl -lcrypto -ldl -lpthread' 'PCRE2_LIBS=-lpcre2-posix -lpcre2-8 -lpcre2-16 -lpcre2-32 -lz -lbz2' DOUBLECONVERSION_LIBS=-ldouble-conversion 'SQLITE_LIBS=-lsqlite3 -lpthread -ldl' 'PSQL_LIBS=-lpq -lpgcommon -lpthread -lz' 'ODBC_LIBS=-lodbc -lodbcinst -lodbccr -lltdl -lpthread -ldl -liconv -lcharset' 'ZSTD_LIBS=-lzstd -lpthread' -I /home/conan/.conan/data/openssl/1.1.1h/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/include -L /home/conan/.conan/data/openssl/1.1.1h/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/lib -I /home/conan/.conan/data/pcre2/10.33/_/_/package/0c7f50fd380a8c3393ca389ff119ec7e28974d9d/include -D PCRE2_STATIC -L /home/conan/.conan/data/pcre2/10.33/_/_/package/0c7f50fd380a8c3393ca389ff119ec7e28974d9d/lib -I /home/conan/.conan/data/double-conversion/3.1.5/_/_/package/da961450d48f172a353d3173ff7efe54d12b811e/include -L /home/conan/.conan/data/double-conversion/3.1.5/_/_/package/da961450d48f172a353d3173ff7efe54d12b811e/lib -I /home/conan/.conan/data/sqlite3/3.31.0/_/_/package/72fab956424a0a2d94c722cc1bfd294beb5cba7c/include -L /home/conan/.conan/data/sqlite3/3.31.0/_/_/package/72fab956424a0a2d94c722cc1bfd294beb5cba7c/lib -I /home/conan/.conan/data/libpq/11.5/_/_/package/f3748ef0523f8bdf5382a8cc0624c2586cc3008e/include -L /home/conan/.conan/data/libpq/11.5/_/_/package/f3748ef0523f8bdf5382a8cc0624c2586cc3008e/lib -I /home/conan/.conan/data/odbc/2.3.7/_/_/package/93d7bafd168ad0429ef9bbb03015255a4f2068f5/include -L /home/conan/.conan/data/odbc/2.3.7/_/_/package/93d7bafd168ad0429ef9bbb03015255a4f2068f5/lib -I /home/conan/.conan/data/zstd/1.4.4/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/include -L /home/conan/.conan/data/zstd/1.4.4/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/lib -I /home/conan/.conan/data/zlib/1.2.11/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/include -L /home/conan/.conan/data/zlib/1.2.11/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/lib -I /home/conan/.conan/data/bzip2/1.0.8/_/_/package/0f605cc66ff048559e2527dae1e478519b0adb51/include -L /home/conan/.conan/data/bzip2/1.0.8/_/_/package/0f605cc66ff048559e2527dae1e478519b0adb51/lib -I /home/conan/.conan/data/libiconv/1.16/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/include -L /home/conan/.conan/data/libiconv/1.16/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/lib -psql_config /home/conan/.conan/data/libpq/11.5/_/_/package/f3748ef0523f8bdf5382a8cc0624c2586cc3008e/bin/pg_config -D_GLIBCXX_USE_CXX11_ABI=0 -xplatform linux-arm-gnueabi-g++ QMAKE_CC=arm-linux-gnueabihf-gcc-7 QMAKE_LINK_C=arm-linux-gnueabihf-gcc-7 QMAKE_LINK_C_SHLIB=arm-linux-gnueabihf-gcc-7 QMAKE_CXX=arm-linux-gnueabihf-g++-7 QMAKE_LINK=arm-linux-gnueabihf-g++-7 QMAKE_LINK_SHLIB=arm-linux-gnueabihf-g++-7
Project ERROR: Cannot run target compiler 'arm-linux-gnueabi-g++'. Output:
===================
===================
Maybe you forgot to setup the environment?
qt/5.15.2@bincrafters/stable: 
qt/5.15.2@bincrafters/stable: ERROR: Package '9782a352f2b22896a551d487d08d69fcf0feb9c7' build failed
qt/5.15.2@bincrafters/stable: WARN: Build folder /home/conan/.conan/data/qt/5.15.2/bincrafters/stable/build/9782a352f2b22896a551d487d08d69fcf0feb9c7
ERROR: qt/5.15.2@bincrafters/stable: Error in build() method, line 663
        self.run("%s/qt5/configure %s" % (self.source_folder, " ".join(args)), run_environment=True)
        ConanException: Error 3 while executing /home/conan/.conan/data/qt/5.15.2/bincrafters/stable/source/qt5/configure -shared -confirm-license -silent -nomake examples -nomake tests -prefix /home/conan/.conan/data/qt/5.15.2/bincrafters/stable/package/9782a352f2b22896a551d487d08d69fcf0feb9c7 -v -opensource -no-gui -no-widgets -release -skip qtsvg -skip qtdeclarative -skip qtactiveqt -skip qtscript -skip qtmultimedia -skip qttools -skip qtxmlpatterns -skip qttranslations -skip qtdoc -skip qtlocation -skip qtsensors -skip qtconnectivity -skip qtwayland -skip qt3d -skip qtimageformats -skip qtgraphicaleffects -skip qtquickcontrols -skip qtserialbus -skip qtserialport -skip qtx11extras -skip qtmacextras -skip qtwinextras -skip qtandroidextras -skip qtwebsockets -skip qtwebchannel -skip qtwebengine -skip qtwebview -skip qtquickcontrols2 -skip qtpurchasing -skip qtcharts -skip qtdatavis3d -skip qtvirtualkeyboard -skip qtgamepad -skip qtscxml -skip qtspeech -skip qtnetworkauth -skip qtremoteobjects -skip qtwebglplugin -skip qtlottie -skip qtquicktimeline -skip qtquick3d --zlib=system -no-opengl -no-vulkan -openssl-linked --glib=no --pcre=system --fontconfig=no --icu=no --sql-mysql=no --sql-psql=yes --sql-odbc=yes --zstd=yes -system-doubleconversion -no-freetype -no-harfbuzz -no-libjpeg -no-libpng -system-sqlite "ZLIB_LIBS=-lz" "OPENSSL_LIBS=-lssl -lcrypto -ldl -lpthread" "PCRE2_LIBS=-lpcre2-posix -lpcre2-8 -lpcre2-16 -lpcre2-32 -lz -lbz2" "DOUBLECONVERSION_LIBS=-ldouble-conversion" "SQLITE_LIBS=-lsqlite3 -lpthread -ldl" "PSQL_LIBS=-lpq -lpgcommon -lpthread -lz" "ODBC_LIBS=-lodbc -lodbcinst -lodbccr -lltdl -lpthread -ldl -liconv -lcharset" "ZSTD_LIBS=-lzstd -lpthread" -I "/home/conan/.conan/data/openssl/1.1.1h/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/include" -L "/home/conan/.conan/data/openssl/1.1.1h/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/lib" -I "/home/conan/.conan/data/pcre2/10.33/_/_/package/0c7f50fd380a8c3393ca389ff119ec7e28974d9d/include" -D PCRE2_STATIC -L "/home/conan/.conan/data/pcre2/10.33/_/_/package/0c7f50fd380a8c3393ca389ff119ec7e28974d9d/lib" -I "/home/conan/.conan/data/double-conversion/3.1.5/_/_/package/da961450d48f172a353d3173ff7efe54d12b811e/include" -L "/home/conan/.conan/data/double-conversion/3.1.5/_/_/package/da961450d48f172a353d3173ff7efe54d12b811e/lib" -I "/home/conan/.conan/data/sqlite3/3.31.0/_/_/package/72fab956424a0a2d94c722cc1bfd294beb5cba7c/include" -L "/home/conan/.conan/data/sqlite3/3.31.0/_/_/package/72fab956424a0a2d94c722cc1bfd294beb5cba7c/lib" -I "/home/conan/.conan/data/libpq/11.5/_/_/package/f3748ef0523f8bdf5382a8cc0624c2586cc3008e/include" -L "/home/conan/.conan/data/libpq/11.5/_/_/package/f3748ef0523f8bdf5382a8cc0624c2586cc3008e/lib" -I "/home/conan/.conan/data/odbc/2.3.7/_/_/package/93d7bafd168ad0429ef9bbb03015255a4f2068f5/include" -L "/home/conan/.conan/data/odbc/2.3.7/_/_/package/93d7bafd168ad0429ef9bbb03015255a4f2068f5/lib" -I "/home/conan/.conan/data/zstd/1.4.4/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/include" -L "/home/conan/.conan/data/zstd/1.4.4/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/lib" -I "/home/conan/.conan/data/zlib/1.2.11/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/include" -L "/home/conan/.conan/data/zlib/1.2.11/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/lib" -I "/home/conan/.conan/data/bzip2/1.0.8/_/_/package/0f605cc66ff048559e2527dae1e478519b0adb51/include" -L "/home/conan/.conan/data/bzip2/1.0.8/_/_/package/0f605cc66ff048559e2527dae1e478519b0adb51/lib" -I "/home/conan/.conan/data/libiconv/1.16/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/include" -L "/home/conan/.conan/data/libiconv/1.16/_/_/package/ef95b2a2795dd6cb7fdaf6af743b1ba1e9dd9e5c/lib" -psql_config "/home/conan/.conan/data/libpq/11.5/_/_/package/f3748ef0523f8bdf5382a8cc0624c2586cc3008e/bin/pg_config" -D_GLIBCXX_USE_CXX11_ABI=0 -xplatform linux-arm-gnueabi-g++ QMAKE_CC="arm-linux-gnueabihf-gcc-7" QMAKE_LINK_C="arm-linux-gnueabihf-gcc-7" QMAKE_LINK_C_SHLIB="arm-linux-gnueabihf-gcc-7" QMAKE_CXX="arm-linux-gnueabihf-g++-7" QMAKE_LINK="arm-linux-gnueabihf-g++-7" QMAKE_LINK_SHLIB="arm-linux-gnueabihf-g++-7"
```

</details>

**Comment by pbeTrifork** [comment]
Confirmed the error is the same using conanio/gcc10-armv7hf

**Comment by ericLemanissier** [comment]
This is https://bugreports.qt.io/browse/QTBUG-73811 and https://bugreports.qt.io/browse/QTBUG-66404 striking again. I'm trying locally to create a custom mkspec for armv7hf. In the meantime, can you try these options: `-o qt:device=linux-arm-generic-g++ -o qt:cross_compile=arm-linux-gnueabihf-` ?

**Comment by ericLemanissier** [comment]
The custom mkspec seems to work. can you please checkout [armhf/5.15.2 branch](https://github.com/bincrafters/conan-qt/tree/armhf/5.15.2) and run `conan create /path/to/conan-qt/working-copy/ qt/5.12.10@bincrafters/stable -o qt:with_icu=False -o qt:GUI=False -o qt:widgets=False -o qt:with_glib=False -o qt:with_mysql=False --build=missing` ?

**Comment by pbeTrifork** [comment]
Well it gets it to compile, however the build doesn't seem to be using hard float any more as I gets `/usr/lib/gcc-cross/arm-linux-gnueabihf/10/../../../../arm-linux-gnueabihf/bin/ld.gold: error: /home/conan/.conan/data/sqlite3/3.29.0/_/_/package/62c65ca87c131481e32d582e10ebd0b82a6df58f/lib/libsqlite3.a(sqlite3.c.o) uses VFP register arguments, output does not` during linking
(With a whole lot of other lines as well as you might imagine)
Confirmed same for gcc7

**Comment by pbeTrifork** [comment]
The armhf/5.15.2 branch works ( but not when trying to create qt/5.12.10 ;) ) both on gcc7hf and gcc10hf
Thank you very much for a very prompt reply.

**Comment by ericLemanissier** [comment]
ok, I integrated the change in 5.15.2 on the stable channel. You just have to wait until it is built and uploaded to bintray.

**Comment by pbeTrifork** [comment]
Will there be any hope of a backport to 5.12.10 or should I just clone the recipes and push it to our own repositories?

**Comment by ericLemanissier** [comment]
I'll try to integrate it later today, but i'm not sure the CI still works on this branch

**Comment by ericLemanissier** [comment]
it should be available on bintray for 5.12.10 too now
